### PR TITLE
c#/11.qnamaker

### PR DESCRIPTION
**Pull request by JasonSowers**
- [x] Is numbered and named correctly and following the spec
--
- [x] All files have been spell checked
- [x] All style cop/ linting issues have been addressed
- [x] Include all DCR changes
- [x] Explicit state save
- [x] Method name changes begin, end etc
- [x] Does not use mainDialog unless that file actually extends Dialog/ ComponentDialog
- [x] Compiles with latest packages
- [x] Runs per instructions in README
- [x] Add commented out code to use blob storage
- [x] Update to look for production/ local (bot configuration)
- [x] onTurnError at adapter
- [x] Include .env or appsettings.json or web.config with botFilePath and botFileSecret
- [x] Deployment scripts folder has been cleaned up and only includes msbot export files
- [x] README to include
- [x] msbot clone steps
- [x] Tools install/ uninstall
- [x] Concise set of steps to run the sample
- [x] C# sample -- Include .bot file in the build output
- [ ] All samples include the latest botfile (no "secretKey" field; replaced with "padLock"). Use latest MSBOT off the tools MyGet feed to generate the bot file
- [x] remove and sort usings
- [x] remove azuredeploy.* under deploymentscripts folder (if any)
- [x] Welcoming users is consistent & correct. See 13.basic bot for nodejs & 3. welcome user for C#

### CODE REVIEW

File: samples/csharp_dotnetcore/11.qnamaker/QnAMaker/QnABot.cs [hunk 1]
```diff
@@ -47,27 +50,63 @@ public QnABot(BotServices services)
...
+                await turnContext.SendActivityAsync($"{turnContext.Activity.Type} event detected", cancellationToken: cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Greet new users as they are added to the conversation.
+        /// </summary>
+        /// <param name="turnContext">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">(Optional) A <see cref="CancellationToken"/> that can be used by other objects
```
Comment (daveta) [inline samples/csharp_dotnetcore/11.qnamaker/QnAMaker/QnABot.cs]: >Optional [](start = 45, length = 8)

Way it's written, actually not optional.

Comment (JasonSowers) [reply samples/csharp_dotnetcore/11.qnamaker/QnAMaker/QnABot.cs]: Nice catch

File: samples/csharp_dotnetcore/11.qnamaker/Startup.cs [hunk 2]
```diff
@@ -62,16 +68,64 @@ public void ConfigureServices(IServiceCollection services)
             services.AddBot<QnABot>(options =>
             {
                 InitCredentialProvider(options);
```
[the remark this answers is not in the document]
Comment (JasonSowers) [reply samples/csharp_dotnetcore/11.qnamaker/Startup.cs]: Was a total oversight on my part.

**Comment by JasonSowers** [comment]
@daveta I need some guidance about what to do with BotServices is there an example out there?

**Pull request merged**